bazel/external/requirements.in

```diff
@@ -1,43 +1,44 @@
 # requirements.in
 setuptools==49.6.0
-grpcio-tools
-grpcio
-redis>=3.5
-protobuf
-six
-pyyaml
-urllib3
-certifi
-sentry-sdk
+grpcio-tools>=1.46.3
+grpcio>=1.47.0
+redis==3.5.3
+protobuf==3.20.1
+six>=1.16.0
+pyyaml==5.3.1
+urllib3==1.26.11
+certifi>=2022.6.15
+sentry-sdk>=1.9.0
 snowflake>=0.0.3
 #  prometheus_client==0.3.1 required (see e.g. magmad/tests/metrics_tests.py)
 prometheus_client==0.3.1
-redis-collections
-python-redis-lock
-jsonpickle
-netifaces
-scapy
-wsgiserver
-flask
-bravado_core
+redis-collections==0.11.0
+python-redis-lock>=3.7.0
+jsonpickle>=2.2.0
+netifaces>=0.11.0
+scapy==2.4.5
+wsgiserver>=1.3
+flask==1.1.4
+markupsafe==1.1.1
+bravado_core==5.17.0
 #  jsonschema version 4 not compatible with magma swagger specs
-jsonschema==3.1.0 
-psutil
-systemd
-cryptography
+jsonschema==3.2.0
+psutil==5.9.1
+systemd-python>=234
+cryptography==3.2.1
 #  h2>=3,<4 is requirement of aioh2 (loaded via bazel)
-h2>=3,<4
+h2==3.2.0
```

Comment:
What is the reason for exactly pinning most packages in requirements.in? As far as I understand the mechanism, `pip-compile` will already create a requirements.txt with exact pinnings, and pip will use requirements.txt to install exactly pinned packages. The requirements.in can then be used to define versions constraints for `pip-compile`.

In this example, the previous version of the line would tell `pip-compile` not to upgrade to `h2 >= 4.0`. With the new line `h2==3.2.0`, `pip-compile` no longer has that information.

That scheme also allows different package managers that don't necessarily have all the packages from pypi. For example a developer could choose to install Python dependencies from nixpkgs on their machine. With the exact pinnings in `requirements.in` this doesn't work anymore as nixpkgs won't have that exact combination of packages available.

Reply:
Sorry, duplicate comment - for some reason Github didn't show my previous comment and I thought it was gone...